src/components/MainNavigation.tsx

```diff
@@ -0,0 +1,160 @@
+import { Link, useNavigate } from 'react-router-dom';
+import logo from '../assets/logo.png';
+import { useState } from 'react';
+import { Button, Menu, MenuItem, Box, Container, ContainerProps, Typography } from '@mui/material';
+import AccountCircleIcon from '@mui/icons-material/AccountCircle';
+import MenuIcon from '@mui/icons-material/Menu';
+import { useAppDispatch, useAppSelector } from '../hooks/redux-hooks';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import httpClient from '../api/httpClient';
+import { UserResponse } from '../types/profile.types';
+import { logout, setProfile } from '../stores/slices/authSlice';
+
+function MainNavigation({ maxWidth }: { maxWidth: ContainerProps['maxWidth'] }) {
+  const isLoggedIn = useAppSelector((state) => state.auth.token) !== null;
+  const userId = localStorage.getItem('sub');
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const queryClient = useQueryClient();
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const { isError } = useQuery({
```

Comment:
Let's create a separate custom hooks to handle these useQuery and useMutation fetches in a separate file.